install/0000_00_cluster-version-operator_01_adminack_configmap.yaml

```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: admin-acks
+  namespace: openshift-cluster-version
+  annotations:
+    release.openshift.io/create-only: "true"
```

Comment:
and our usual `include.release.openshift.io/self-managed-high-availability: "true"
` profile annotation?

Reply:
In the enhancement we said we were not going to have CVO manage this one in order to allow an admin to simply delete it.

Reply:
In this PR's [e2e run][1]:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/633/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1423306540869226496/artifacts/e2e-agnostic/gather-extra/artifacts/configmaps.json | jq -r '.items[].metadata | select(.name == "admin-acks")'
{
  "annotations": {
    "release.openshift.io/create-only": "true"
  },
  "creationTimestamp": "2021-08-05T15:56:27Z",
  "name": "admin-acks",
  "namespace": "openshift-cluster-version",
  "resourceVersion": "68",
  "uid": "d611d1f7-ba10-46b0-a0e9-9e4f2b4a4abf"
}
```

So yeah, something seems to be creating this manifest...  The CVO loads manifests [from two directories][2], [`manifests` for CVO manifests, and `release-manifests` for everyone else's manifests][3].  Both directories are [filtered by profile][4].  15:56:27Z is very early in the install, while [the installer is still creating Terraform resources][5]:

```
time="2021-08-05T15:59:28Z" level=info msg="Waiting up to 20m0s for the Kubernetes API at https://api.ci-op-345d5w3h-3302f.ci.azure.devcluster.openshift.com:6443..."
time="2021-08-05T15:59:28Z" level=info msg="API v1.21.1-1423+8268f88fb9294f-dirty up"
```

Poking at audit logs:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/633/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1423306540869226496/artifacts/e2e-agnostic/gather-audit-logs/artifacts/audit-logs.tar | tar xz --strip-components=1
$ zgrep -h '"resource":"configmaps"' audit_logs/*/*.gz 2>/dev/null | jq -r 'select(.objectRef.name // "" | contains("admin-"))  | .stageTimestamp + " " + (.responseStatus.code | tostring) + " " + .verb + " " + .objectRef.name + " " + .user.username' | sort
2021-08-05T16:09:44.035961Z 200 get admin-gates system:serviceaccount:kube-system:generic-garbage-collector
2021-08-05T16:13:56.999908Z 200 get admin-gates system:serviceaccount:kube-system:generic-garbage-collector
...
2021-08-05T16:44:03.026572Z 200 get admin-gates system:serviceaccount:openshift-cluster-version:default
2021-08-05T16:50:50.925720Z 200 get admin-gates system:serviceaccount:openshift-cluster-version:default
```

So I guess we don't have audit logs for stuff that happened on the bootstrap host.  Hmm, I wonder if we account for cluster-profile when [rendering our manifests on the bootstrap host][6].  We're [picking up the env var][7].  But then [we are only using it for templating, not for filtering][8].  So we need to get a `shouldExclude` call somewhere under `Render`.

And while leaving off profile annotations will work on bootstrap with the current rendering code, it will not work on updates from previous versions to this new version (which we should be able to see in CI here), because outside of the rendered manifests, the CVO will ignore this profile-less file.

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/633/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1423306540869226496
[2]: https://github.com/openshift/cluster-version-operator/blob/77a24f77b8f6e9975b5f41a787d6368ff609e391/pkg/payload/payload.go#L301-L309
[3]: https://github.com/openshift/cluster-version-operator/blob/77a24f77b8f6e9975b5f41a787d6368ff609e391/pkg/payload/payload.go#L93-L94
[4]: https://github.com/openshift/cluster-version-operator/blob/77a24f77b8f6e9975b5f41a787d6368ff609e391/pkg/payload/payload.go#L182-L189
[5]: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/633/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1423306540869226496/artifacts/e2e-agnostic/ipi-install-install/artifacts/.openshift_install.log
[6]: https://github.com/openshift/installer/blob/09d231228f974e00a8c18946add3a3ff32fbd2b6/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template#L80-L105
[7]: https://github.com/openshift/cluster-version-operator/blob/77a24f77b8f6e9975b5f41a787d6368ff609e391/cmd/render.go#L43-L54
[8]: https://github.com/openshift/cluster-version-operator/blob/77a24f77b8f6e9975b5f41a787d6368ff609e391/pkg/payload/render.go#L17-L105

Reply:
IIUC seems the safer way to go is to go ahead and include the `self-managed-high-availability: "true" `. The only "side-affect" is that if an admin decides to delete the configmap CVO will re-create it in its original form but there's really no reason to delete it. If the gates don't apply the code never checks admin-acks and if any gates do apply removing the configmap still results in Upgradeable=false with an admin-acks NotFound error.

Reply:
@jottofar Were you able to confirm if the manifests in cvo/install directory goes in to the CVO manifest directory?

Reply:
Will making it CVO managed can cause any issue when we move from 4.7 to 4.8?

Reply:
> Were you able to confirm if the manifests in cvo/install directory goes in to the CVO manifest directory?

I will check but they should end up in the same place as the existing CVO manifests in `install`.

>  Will making it CVO managed can cause any issue when we move from 4.7 to 4.8?

I can't think of any issue. It's a new manifest that comes in as part of some 4.8.z and 4.9 and the annotation simply causes CVO to NOT ignore it when`sync'ing`. The CI upgrade test has passed before but I'll look closer at it. Is there something specific you have in mind or are you thinking there's something different about 4.7->4.8 vs 4.8->4.9?

Reply:
> Were you able to confirm if the manifests in cvo/install directory goes in to the CVO manifest directory?

Confirmed that the two new configmap manifests are in /manifests directory of the CVO pod.

Reply:
> Will making it CVO managed can cause any issue when we move from 4.7 to 4.8?

@LalatenduMohanty see comment/question regarding this above